project/Settings.scala

```diff
@@ -5,7 +5,7 @@ import sbtassembly.AssemblyKeys._
 object Settings {
   val buildName = "hazelcast-spark"
   val buildVersion = "0.1-SNAPSHOT"
-  val buildScalaVersion = "2.10.5"
+  val buildScalaVersion = "2.11.8"
```

Comment:
From Spark documentation : Spark 1.6.2 uses Scala 2.10. You will need to use a compatible Scala version (2.10.x).